server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1412,12 +1412,40 @@ private void closeConsistencyCheck() {
 
     if (!otherLogs.isEmpty() || !currentLogs.isEmpty() || !referencedLogs.isEmpty()) {
       String msg = "Closed tablet " + extent + " has walog entries in memory currentLogs = "
-          + currentLogs + "  otherLogs = " + otherLogs + " refererncedLogs = " + referencedLogs;
+          + currentLogs + "  otherLogs = " + otherLogs + " referencedLogs = " + referencedLogs;
       log.error(msg);
       throw new RuntimeException(msg);
     }
 
-    // TODO check lastFlushID and lostCompactID - ACCUMULO-1290
+    // If a table hasn't been flushed before it was closed then lastFlushID will be -1 while
+    // getFlushID will be 0.
```

Comment:
I'm not sure that this comment is correct. In fact, I'm not sure that the original comment from 2013 applies at this point. From what my IDE search found, ZTABLE_FLUSH_ID is only set in [ManagerClientServiceHandler#initiateFlush](https://github.com/apache/accumulo/blob/main/server/manager/src/main/java/org/apache/accumulo/manager/ManagerClientServiceHandler.java#L125) which IIRC is used when someone submits a `flush` command in the shell. From what I can tell, if a Tablet is hosted on a TabletServer in the normal course of business and minor compactions occur naturally, then `lastFlushID` will be equal to the value passed in the Tablet constructor and `flushID` will be equal to the value stored in ZooKeeper.

Reply:
That could very well be true. I based the comments off of the information that was provided in the original issue on GitHub. I can remove this if it is inaccurate or outdated or change the wording.

Reply:
I would like someone else to confirm what I'm seeing. There may be a different way to do the close consistency check for flushes and compactions, I just think the information in the original issue, which this change is based off of, is out of date.

Reply:
@dlmarion Is this, and your other comment below resolved with the updated changes? Is this PR ready to be merged in your opinion?

Reply:
From what I recall when I moved the issue into GitHub, I did notice certain circumstances which led to lastFlushID and getFlushID showing that no flush has occurred but still not being consistent with each other. I will re-investigate to see if that still holds true.

Reply:
From just a quick test run with accumulo-testing ingesting and things flushing normally (no compactions happened yet), I saw lastFlushID 0, last CompactID -1, getCompactID optionalLong empty, and getFlushID opt long 0.  The states match up but I am not sure the consistency checks here will pass through consistently.